usaspending_api/common/helpers/spark_helpers.py

```diff
@@ -534,6 +548,72 @@ def log_hadoop_config(spark: SparkSession, config_key_contains=""):
     ]
 
 
+def load_dict_to_delta_table(spark, s3_data_bucket, table_schema, table_name, data, overwrite=False):
+    table_to_col_names_dict = {}
+    table_to_col_names_dict["transaction_fabs"] = TRANSACTION_FABS_COLUMNS
+    table_to_col_names_dict["transaction_fpds"] = TRANSACTION_FPDS_COLUMNS
+    table_to_col_names_dict["transaction_normalized"] = list(TRANSACTION_NORMALIZED_COLUMNS)
+    table_to_col_names_dict["awards"] = list(AWARDS_COLUMNS)
+    table_to_col_names_dict["financial_accounts_by_awards"] = list(FINANCIAL_ACCOUNTS_BY_AWARDS_COLUMNS)
+
+    table_to_col_info_dict = {}
+    for tbl_name, col_info in zip(
+        ("transaction_fabs", "transaction_fpds"), (TRANSACTION_FABS_COLUMN_INFO, TRANSACTION_FPDS_COLUMN_INFO)
+    ):
+        table_to_col_info_dict[tbl_name] = {}
+        for col in col_info:
+            table_to_col_info_dict[tbl_name][col.dest_name] = col
+
+    # Make sure the table has been created first
+    call_command(
+        "create_delta_table",
+        "--destination-table",
+        table_name,
+        "--alt-db",
+        table_schema,
+        "--spark-s3-bucket",
+        s3_data_bucket,
+    )
+
+    if data:
```

Comment:
Would it be helpful to get a list of pyspark `Row` objects?
You can create `Row` objects from dict by [unpacking the dict](https://stackoverflow.com/a/38253641).

```
from pyspark.sql import Row

Row(**data_as_dict)
```

Reply:
Are you thinking it might be better to convert the list of dictionaries to a list of Row objects then use a Spark function to insert the rows?

If so, we may still need some logic to cleanup some fields such as converting `date`/`datetime` fields to `isoformat()` or using `col_info` if available to set defaults for empty values.

Lots of the logic for building the INSERT SQL could be removed though

Reply:
Yeah, fine to leave as is for now if it's working.